src/contractor/contractor.cpp

```diff
@@ -81,6 +81,41 @@ inline EdgeWeight distanceAndSpeedToWeight(double distance_in_meters, double spe
     return std::max<EdgeWeight>(1, static_cast<EdgeWeight>(std::round(duration * 10)));
 }
 
+// Returns updated edge weight
+template <class IterType>
+EdgeWeight getNewWeight(IterType speed_iter,
+                        const double &segment_length,
+                        const std::vector<std::string> &segment_speed_filenames,
+                        EdgeWeight oldWeight,
+                        const double log_edge_updates_factor)
+{
+    auto new_segment_weight =
+        (speed_iter->speed_source.speed > 0)
+            ? distanceAndSpeedToWeight(segment_length, speed_iter->speed_source.speed)
+            : INVALID_EDGE_WEIGHT;
+    // the check here is enabled by the `--edge-weight-updates-over-factor` flag
+    // it logs a warning if the new weight exceeds a heuristic of what a reasonable weight update is
+    if (log_edge_updates_factor > 0 && new_segment_weight != 0)
+    {
+        auto newSecs = new_segment_weight / 10.0;
+        auto oldSecs = oldWeight / 10.0;
+        auto approx_original_speed = (segment_length / oldSecs) * 3.6;
+        if ((new_segment_weight < oldWeight) && ((oldSecs / newSecs) >= log_edge_updates_factor))
```

Comment:
Why

> (new_segment_weight < oldWeight) && ((oldSecs / newSecs) >= log_edge_updates_factor)

- Why checking for both weight and speed? Can't we simply log cases where `new weight > old weight * f`?
- Can `newSecs` ever be zero resulting in a div-by-zero crash here?